ml-algorithms/src/main/java/org/opensearch/ml/engine/algorithms/remote/ConnectorUtils.java

```diff
@@ -245,4 +258,47 @@ public static SdkHttpFullRequest signRequest(
 
         return signer.sign(request, params);
     }
+
+    public static SdkHttpFullRequest buildSdkRequest(
+        Connector connector,
+        Map<String, String> parameters,
+        String payload,
+        SdkHttpMethod method
+    ) throws Exception {
+        String endpoint = connector.getPredictEndpoint(parameters);
+        URL url = new URL(endpoint);
+        String protocol = url.getProtocol();
+        String host = url.getHost();
+        int port = url.getPort();
+        MLHttpClientFactory.validate(protocol, host, port);
+        String charset = parameters.getOrDefault("charset", "UTF-8");
+        RequestBody requestBody;
+        if (payload != null) {
+            requestBody = RequestBody.fromString(payload, Charset.forName(charset));
+        } else {
+            requestBody = RequestBody.empty();
+        }
+        if (SdkHttpMethod.POST == method && "0".equals(requestBody.optionalContentLength().get().toString())) {
```

Comment:
`"0".equals(requestBody.optionalContentLength().get().toString())` --> Why do we need to convert to string to check?

Reply:
if `RequestBody.empty()`, what do we get from `requestBody.optionalContentLength().get()`?

Reply:
> `"0".equals(requestBody.optionalContentLength().get().toString())` --> Why do we need to convert to string to check?

The toString() can be removed.

Reply:
> if `RequestBody.empty()`, what do we get from `requestBody.optionalContentLength().get()`?

it's 0.